app/src/main/kotlin/com/hippo/ehviewer/ui/tools/Dialog.kt

```diff
@@ -156,13 +156,17 @@ class DialogState {
         hint: String? = null,
         isNumber: Boolean = false,
         @StringRes confirmText: Int = android.R.string.ok,
+        onDismiss: (() -> Unit)? = null,
```

Comment:
`onUserDismiss` as it only triggered by user action not coroutine cancellation

Reply:
Maybe just use the same name as onDismissRequest

Reply:
> Maybe just use the same name as onDismissRequest

Nope, as the dialog is definitely dismissed when it triggered, we must avoid semantic clash.